### PR TITLE
Adds postgres configuration steps

### PR DESCRIPTION
**Pull request by lomky**
I rule and have completed some work on Case Manager that's ready for review!

For folks installing without docker who are using Postgres for the first time, there are some steps we have to follow to get configured. 

The two possible paths are:
 - setting up a postgres role for your system user, setting those values in your ENV, and tweaking the `config/database.yml` to use them ([inexact reference](https://www.digitalocean.com/community/tutorials/how-to-use-postgresql-with-your-ruby-on-rails-application-on-ubuntu-18-04))
 - tweaking your postgres server to not require passwords by adding `trust` methods to `pg_hba.conf` 

I went with the first as it feels more user friendly to tweak your local env vs the database settings.

This pull request makes the following changes:
* adds user & pass to `config/database.yml`
* adds config instructions to `docs/SETUP.md`

For reviewer:
* Adjust the title to explain what it does for the notification email to the listserv.
* Tag this PR:
  * `feature` if it contains a feature, fix, or similar. This is anything that contains a user-facing fix in some way, such as frontend changes, alterations to backend behavior, or bug fixes.
  * `dependencies` if it contains library upgrades or similar. This is anything that upgrades any dependency, such as a Gemfile update or npm package upgrade.
* If it contains neither, no need to tag this PR.

### CODE REVIEW

File: docs/SETUP.md [hunk 1]
```diff
@@ -70,6 +70,25 @@ We use Postgres around these parts. Installation will differ based on your OS.
...
+   ```
+   sudo -u postgres createuser -s `whoami`
+   ```
+ * Set your env variables:
+   * Mac:
+     ```
+     echo "export POSTGRES_USER=`whoami`" >>~/.bashrc
+     echo "export POSTGRES_PASSWORD=postgres" >>~/.bashrc
+     echo "export PGDATABASE=dcaf_case_management_development" >>~/.bashrc
+     ```
```
Comment (lomky) [inline docs/SETUP.md:84]: Feedback requested from a Mac user: do these steps result in the values appearing in your ENV on next terminal start up? If not, which file should they go in?

File: docs/SETUP.md [hunk 2]
```diff
@@ -70,6 +70,25 @@ We use Postgres around these parts. Installation will differ based on your OS.
...
+   * Ubuntu:
+     ```
+     echo "export POSTGRES_USER=`whoami`" >>~/.bashrc
+     echo "export POSTGRES_PASSWORD=postgres" >>~/.bashrc
```
Comment (lomky) [inline docs/SETUP.md:88]: These two make the Rails server happy

File: docs/SETUP.md [hunk 3]
```diff
@@ -70,6 +70,25 @@ We use Postgres around these parts. Installation will differ based on your OS.
...
+     echo "export PGDATABASE=dcaf_case_management_development" >>~/.bashrc
```
Comment (lomky) [inline docs/SETUP.md:89]: This one makes your `psql` command work without having to pass `-d dcaf_case_management_development`

File: docs/SETUP.md (hunk 1 above)
Comment (colinxfleming) [reply docs/SETUP.md:84]: Macs default to zsh now, obnoxiously, which makes it tough to write directions like this. But the tldr is that I think if you are running a recent mac os your default shell rc file is going to be `~/.zshrc` so I think we prrrrrrrob want to point that in that direction.

File: docs/SETUP.md (hunk 3 above)
Comment (colinxfleming) [reply docs/SETUP.md:89]: `rails db` also works for this fwiw.

This doesn't interact with the test env, right? The `database.yml` overrules it?

File: docs/SETUP.md [hunk 4]
```diff
@@ -70,6 +70,25 @@ We use Postgres around these parts. Installation will differ based on your OS.
...
 
+* Configure Postgres
+ * Setup your user; the suggested password is `postgres` if you only use this machine for development purposes
+   ```
+   sudo -u postgres createuser -s `whoami`
+   ```
+ * Set your env variables:
+   * Mac:
+     ```
+     echo "export POSTGRES_USER=`whoami`" >>~/.bashrc
```
Comment (colinxfleming) [inline docs/SETUP.md:81]: I wonder if `dotenv` might save us some kinda particular/confusing shell mucking by letting us just stick this in an `.env` file in the root of the project? https://github.com/bkeepers/dotenv I will defer to you whether or not it's worth the dependency add.

**Review by colinxfleming — COMMENTED** [review]
couple of notes and a question. I'm feelin' it - this is great work sanding the rough edges of our new setup. Interested in your thoughts on the comments. Thanks much @lomky !

File: docs/SETUP.md (hunk 4 above)
Comment (mdworken) [reply docs/SETUP.md:81]: I personally like dotenv a lot.

File: config/database.yml [hunk 5]
```diff
@@ -5,6 +5,8 @@ development:
   url: <%= ENV.fetch('pg_url', 'postgres://localhost') %>
   database: dcaf_case_management_development
   pool: 5
+  username: <%= ENV['POSTGRES_USER'] || `whoami` %>
```
Comment (mdworken) [inline config/database.yml:8]: I'd default this to `postgres` unless there's a specific reason not to.

Comment (lomky) [reply config/database.yml:8]: Could you say why?

**Comment by lomky** [comment]
Finally bit the bullet and installed from scratch now that we have postgres - this now represents the functional steps and settings for a bare metal install without having to touch anything outside of the installed repo! ready for final look over

**Review by colinxfleming — APPROVED** [review]
thanks very much for seeing this thru!

**Pull request merged**